packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -0,0 +1,389 @@
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class OnsetPicker extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    // This binding is necessary to make `this` work in the callback
+    this.changeGranularity = this.changeGranularity.bind(this);
+    this.changePrecision = this.changePrecision.bind(this);
+    this.changeAge = this.changeAge.bind(this);
+    this.changeAgeUnit = this.changeAgeUnit.bind(this);
+    this.changeYear = this.changeYear.bind(this);
+    this.changeMonth = this.changeMonth.bind(this);
+    this.changeDate = this.changeDate.bind(this);
+    this.handleDateChange = this.handleDateChange.bind(this);
+
+    const ageValues = OnsetUtils.onsetToAge(this.props.birthdate, this.props.onsetDate);
+
+    this.state = {
+      granularity: this.props.granularity,
+      precision: this.props.precision,
+      onsetDate: this.props.onsetDate ? moment(this.props.onsetDate) : moment(),
+      age: ageValues.age,
+      ageUnit: ageValues.ageUnit,
+    };
+  }
+
+  /*
+   * Change state for granularity
+   */
+  changeGranularity(event) {
+    if (event.target.value === GranularityOptions.AGE) { // Calculate age values and update onsetDate to match age calculation
+      const ageValues = OnsetUtils.onsetToAge(this.props.birthdate, this.state.onsetDate);
+      const newDate = moment(this.props.birthdate).add(ageValues.age, ageValues.ageUnit);
+
+      this.setState({
+        age: ageValues.age,
+        ageUnit: ageValues.ageUnit,
+        onsetDate: newDate,
+      });
+
+      this.handleDateChange(newDate);
+    }
+    this.setState({ granularity: event.target.value });
+
+    if (this.props.granularitySelectOnChange) {
+      this.props.granularitySelectOnChange(event.target.value);
+    }
+  }
+
+  /*
+   * Change state for precision
+   */
+  changePrecision(event) {
+    this.setState({ precision: event.target.value });
+
+    if (this.props.precisionSelectOnChange) {
+      this.props.precisionSelectOnChange(event.target.value);
+    }
+  }
+
+  /*
+   * Change state for the age when using AGE granularity, and update onset date
+   */
+  changeAge(event) {
+    const newDate = moment(this.props.birthdate).add(Number(event.target.value), this.state.ageUnit);
+
+    this.setState({
+      age: Number(event.target.value),
+      onsetDate: newDate,
+    });
+    this.handleDateChange(newDate);
+  }
+
+  /*
+   * Change state for duration when using age granularity, and update onset date
+   */
+  changeAgeUnit(event) {
+    const newDate = moment(this.props.birthdate).add(this.state.age, event.target.value);
+
+    this.setState({
+      ageUnit: event.target.value,
+      onsetDate: newDate,
+    });
+    this.handleDateChange(newDate);
+  }
+
+  /*
+   * Update onset date when year changes
+   */
+  changeYear(event) {
+    let newDate = moment(this.state.onsetDate).year(event.target.value);
```

Comment:
You would want to update this since this.state.onsetDate is now a moment object.

Reply:
No longer wrapping a moment object as a moment object.